addons/binding/org.openhab.binding.toon/src/main/java/org/openhab/binding/toon/handler/ToonBridgeHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.toon.config.ToonBridgeConfiguration;
+import org.openhab.binding.toon.internal.ToopApiClient;
+import org.openhab.binding.toon.internal.api.ToonState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ToonBridgeHandler} class connects bridges the Toon api and connected displays .
+ *
+ * @author Jorg de Jong - Initial contribution
+ */
+public class ToonBridgeHandler extends BaseBridgeHandler {
+    private Logger logger = LoggerFactory.getLogger(ToonBridgeHandler.class);
+    private ToonBridgeConfiguration configuration;
+    private ToopApiClient apiClient;
+
+    protected ScheduledFuture<?> refreshJob;
+
+    public ToonBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Toon API bridge handler.");
+        updateStatus(ThingStatus.INITIALIZING);
+
+        configuration = getConfigAs(ToonBridgeConfiguration.class);
+        logger.debug("refresh interval {}", configuration.refreshInterval);
+
+        disposeApiClient();
+        apiClient = new ToopApiClient(configuration);
+
+        updateStatus();
+        startAutomaticRefresh();
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.cancel(true);
+    }
+
+    private void startAutomaticRefresh() {
+        refreshJob = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                updateChannels();
+            }
+        }, 1, configuration.refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private void updateChannels() {
+        if (getThing().getThings().isEmpty()) {
+            return;
+        }
+        logger.debug("updateChannels");
+        try {
+            ToonState state;
+            try {
+                state = apiClient.collect();
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                return;
+            }
+
+            if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+            for (Thing handler : getThing().getThings()) {
+                ThingHandler thingHandler = handler.getHandler();
+                if (thingHandler instanceof AbstractToonHandler) {
+                    AbstractToonHandler moduleHandler = (AbstractToonHandler) thingHandler;
+                    moduleHandler.updateChannels(state);
+                }
+            }
+        } catch (Exception e) {
+            logger.debug("updateChannels acting up", e);
+        }
+    }
+
+    private void disposeApiClient() {
+        if (apiClient != null) {
+            apiClient.logout();
+        }
+        apiClient = null;
+    }
+
+    @Override
+    public void handleRemoval() {
```

Comment:
I assume you rather wanted to override dispose() here - handleRemoval is called, if a Thing is physically deleted (by the user), dispose() is called, whenever this handler instance is shut down.